rotkehlchen/tests/unit/decoders/test_convex.py

```diff
@@ -0,0 +1,1171 @@
+from rotkehlchen.chain.ethereum.decoding.constants import CPT_GAS
+from rotkehlchen.chain.ethereum.modules.convex.constants import CONVEX_POOLS, CPT_CONVEX
+from rotkehlchen.chain.ethereum.modules.convex.decoder import BOOSTER
+from rotkehlchen.chain.ethereum.structures import EthereumTxReceipt, EthereumTxReceiptLog
+from rotkehlchen.chain.ethereum.types import string_to_ethereum_address
+from rotkehlchen.chain.ethereum.utils import multicall
+from rotkehlchen.constants import ONE, ZERO
+from rotkehlchen.constants.assets import A_CRV, A_CVX, A_ETH
+from rotkehlchen.constants.ethereum import EthereumConstants
+from rotkehlchen.db.ethtx import DBEthTx
+from rotkehlchen.fval import FVal
+from rotkehlchen.types import EthereumTransaction, Location, deserialize_evm_tx_hash
+from rotkehlchen.utils.hexbytes import hexstring_to_bytes
+from rotkehlchen.utils.misc import hex_or_bytes_to_address
+
+
+def test_convex_pools(ethereum_manager):
+    booster_contract = EthereumConstants.contract('CONVEX_BOOSTER')
+    pools_count = booster_contract.call(
+        ethereum=ethereum_manager,
+        method_name='poolLength',
+    )
+    calls_to_booster = []
+    for i in range(pools_count):
+        calls_to_booster.append(
+            (
+                booster_contract.address,
+                booster_contract.encode('poolInfo', [i]),
+            ),
+        )
+    booster_result = multicall(
+        ethereum=ethereum_manager,
+        calls=calls_to_booster,
+    )
+    convex_rewards_addrs = []
+    convex_lp_tokens_addrs = []
+    lp_tokens_contract = EthereumContract(  # only need it to encode and decode
+        address=ZERO_ADDRESS,
+        abi=EthereumConstants.abi('CONVEX_LP_TOKEN'),
+        deployed_block=0,
+    )
+    for single_booster_result in booster_result:
+        lp_token_addr = hex_or_bytes_to_address(single_booster_result[0:32])
+        crv_rewards = hex_or_bytes_to_address(single_booster_result[3 * 32:4 * 32])
+        convex_rewards_addrs.append(crv_rewards)
+        convex_lp_tokens_addrs.append(lp_token_addr)
+
+    # We query this info from chain instead of using data from our assets database since
+    # if convex adds a new pool with new lp token we won't know its properties (because it won't
+    # be in our DB)
+    calls_to_lp_tokens = []
+    for lp_token_addr in convex_lp_tokens_addrs:
+        calls_to_lp_tokens.append((lp_token_addr, lp_tokens_contract.encode('symbol')))
+
+    lp_tokens_result = multicall(
+        ethereum=ethereum_manager,
+        calls=calls_to_lp_tokens,
+    )
+
+    queried_convex_pools_info = {}
+    for convex_reward_addr, single_lp_token_result in zip(convex_rewards_addrs, lp_tokens_result):
+        decoded_lp_token_result = lp_tokens_contract.decode(single_lp_token_result, 'symbol')
+        queried_convex_pools_info[convex_reward_addr] = decoded_lp_token_result[0]  # pylint: disable=unsubscriptable-object  # noqa: E501
+
+    assert CONVEX_POOLS == queried_convex_pools_info
```

Comment:
instead of asserting use the same mechanism we use in asset detection tests for exchanges. It should emit warnings when things are missing. Otherwise we will have failing tests every time a new pool is added.

Reply:
Ok